app/views/registrant/domains/partials/_contacts.html.erb

```diff
@@ -0,0 +1,40 @@
+<% panel_class = errors ? 'panel-danger' : 'panel-default' %>
+<div class="<%= panel_class %> panel">
+    <div class="panel-heading">
+        <%= header %>
+    </div>
+
+    <div class="table-responsive">
+        <table class="table table-hover table-bordered table-condensed">
+            <thead>
+                <tr>
+                    <th class="col-xs-4"><%= Contact.human_attribute_name :name %></th>
+                    <th class="col-xs-4"><%= Contact.human_attribute_name :code %></th>
+                    <th class="col-xs-4"><%= Contact.human_attribute_name :email %></th>
+                </tr>
+            </thead>
+
+            <tbody>
+                <% contacts.each do |contact| %>
+                    <tr>
+                        <td>
+                            <%= link_to contact, registrant_domain_contact_path(domain, contact) %>
```

Comment:
Why providing a custom implementation of `to_s` for `Contact` class is a bad thing and what exactly is prone to error given we have tests for that part?

Reply:
It is prone to error because as being implicit, it relies on undocumented behaviour.
-None of this is publicly documented, neither in examples for `link_for` or not in anything that goes deeper than that. 
This is where this comes from: https://github.com/rails/rails/blob/v4.2.9/actionview/lib/action_view/helpers/tag_helper.rb#L146

In Rails 5.2, that is changed quite a lot: https://github.com/rails/rails/commit/a65a3bde0bee7d3d28c30775ec955260ed61e96c#diff-2eb399a01976bca6aad9e85aa67b403d

On top of everything, none of the actual tag helper tests test for this scenario (passing an object instead of a string to `a` tag. https://github.com/rails/rails/blob/master/actionview/test/template/tag_helper_test.rb

---
Where's the test for this link? Or do you mean the fact that we do have a test for `Contact#to_s`?